src/main/java/io/vinyldns/java/model/zone/Zone.java

```diff
@@ -223,6 +237,7 @@ public boolean equals(Object o) {
     if (!adminGroupId.equals(zone.adminGroupId)) return false;
     if (latestSync != null ? !latestSync.equals(zone.latestSync) : zone.latestSync != null)
       return false;
+    if (!backendId.equals(zone.backendId)) return false;
```

Comment:
Think we need to do a `null` check here since `backendId` is optional: `if (backendId != null ? !backendId.equals(zone.backendId) : zone.backendId != null) return false;`